h2o-algos/src/test/java/hex/tree/gbm/GBMTest.java

```diff
@@ -4183,4 +4184,92 @@ public void testGBMFeatureInteractions() {
       Scope.exit();
     }
   }
+
+  @Test
+  public void testGBMFeatureInteractionsGainTest() {
+    Frame fr=null;
+    GBMModel model = null;
+    Scope.enter();
+    try {
+      Frame f = Scope.track(parse_test_file("smalldata/logreg/prostate.csv"));
+      f.replace(f.find("CAPSULE"), f.vec("CAPSULE").toNumericVec());
+      DKV.put(f);
+
+      GBMModel.GBMParameters parms = makeGBMParameters();
+      parms._response_column = "CAPSULE";
+      parms._train = f._key;
+      parms._ntrees = 1;
+
+      model = new GBM(parms).trainModel().get();
+
+      FeatureInteractions featureInteractions = model.getFeatureInteractions(2,100,-1);
+
+      SharedTreeSubgraph treeSubgraph = model.getSharedTreeSubgraph(0, 0);
+
+      String[] keysToCheck = new String[]{"DPROS", "ID", "PSA", "GLEASON", "VOL"};
+      for (String feature : keysToCheck) {
+        if (!feature.equals(parms._response_column)) {
+          List<SharedTreeNode> featureSplits = treeSubgraph.nodesArray.stream()
+                  .filter(node -> feature.equals(node.getColName()))
+                  .collect(Collectors.toList());
+
+          double featureGain = 0.0;
+          for (int i = 0; i < featureSplits.size(); i++) {
+            SharedTreeNode currSplitNode = featureSplits.get(i);
+            featureGain += currSplitNode.getSquaredError() - currSplitNode.getLeftChild().getSquaredError() - currSplitNode.getRightChild().getSquaredError();
+          }
+
+          assertEquals(featureGain, featureInteractions.get(feature).gain, 0.0001);
+        }
+      }      
+      
+      DKV.remove(f._key);
```

Comment:
Why?

Reply:
not needed since already tracked by scope, will remove

Reply:
cc7b119